### ISSUE TITLE
Docs need a complete example of nested validation

### ISSUE THREAD

**Issue by ocratravis** [vuelidate/vuelidate#1062]
Suppose that I have a parent component, containing this:

```
<template>
    <div>

        <input type="text" v-model="v$.companyName.$model">
        <ul v-if="v$.companyName.$error" class="form-errors">
            <li v-for="$error in v$.companyName.$errors">
                {{ $error.$message }}
            </li>
        </ul>

        <EmployeeFields
            v-for="employee in v$.employees"
        />

        <button @click="save">Save Company</button>

    </div>
</template>

<script setup>

import { reactive } from 'vue'
import useVuelidate from '@vuelidate/core'
import { required } from '@vuelidate/validators'
import EmployeeFields from '@/components/EmployeeFields.vue'

const formState = reactive({
    companyName: '',
    employees: [
        { id: 1, name: 'Bob', email: 'bob@example.com' },
        { id: 2, name: 'Sue', email: 'sue@example.com' },
    ]
})

const formRules = {
    companyName: { required },
    employees: {
        // TODO
    }
}

const v$ = useVuelidate(formRules, formState)

function save() {

    // TODO: submit the entire form to an API

}

</script>
```

I'm trying to figure out how to validate the employees array.

The docs for "[Validating collections](https://vuelidate-next.netlify.app/advanced_usage.html#validating-collections)" recommend using [nested validations](https://vuelidate-next.netlify.app/advanced_usage.html#nested-validations) in this case.

So I've read those docs, as well. But I'm afraid that I simply don't understand the example code. It doesn't provide enough information.

Specifically, I'm wondering:

* What does the code inside CompA and CompB look like?
* Typically – as shown in my code above – the parent component is responsible for maintaining the entire formState, and for submitting it to an API. However, the nested validation example in the docs seems to say that the formState and formRules for nested data should live entirely inside the child component. If that's the case, then how should the parent component _access_ that data, so it can submit it to the API?

Basically: it would really help me to see a complete example of nested validation. The one in the docs is too pared down.

Thanks!

**Comment by dobromir-hristov** [comment]
Hey there, sorry for the confusion. Yes the docs need some love.

As for your question, think of it like this. Your state can now be kept in the parent or the child, its up to you.

Each Vuelidate instance inside a component will collect any available child Vuelidate instances, without you doing anything.

So if you have a component, now matter how deep, that also has a Vuelidate instance, by default, if that instance has an error, the parent Vuelidate instance also has an error. This way you can just do `await v$.$validate()` on the parent and you know whether any of the children is busted or not.

Now ofc, if the parent is to send out the data, then yes, it would be the easiest to keep the data on it and just pass it down to the children via props (or however else you want, can be Vuex).

You can find some examples here, which I agree I should clean up as well - https://github.com/vuelidate/vuelidate/blob/next/packages/test-project/src/components/NestedValidations.vue

**Comment by twei55** [comment]
I have exactly the same issue and agree with @ocratravis that it would be great to have some more information on nested validations in the docs, especially some example code for a nested component.

**Comment by ocratravis** [comment]
If anyone actually understands how to validate collections with Vuelidate, _please_ add it to the Vuelidate docs. You would be my hero.

I've tried every approach mentioned in the docs – nested validations, `ValidateEach`, `helpers.forEach()` – and nothing quite works.

I can _almost_ get it working – but in every case, the validation messages get populated into `$silentErrors`, not `$errors`. And I can't find anything in the docs which explains exactly what `$silentErrors` _is_, or how it works.

I do have one page where nested validation is working (_i.e._, the errors get populated into `$errors`) – but I have no idea _why_ it works on this page. I've tried to replicate the same approach on another page, and it doesn't work.

I've been working with Vuelidate for a year and a half, and I still can't figure out what I'm supposed to be doing here. I don't know if I'm misunderstanding something, or if there are bugs in Vuelidate, or what.
